### PR TITLE
Fix startup page

### PR DESCRIPTION
**Pull request by aadcg**
# Description

Fix UI styling.

Fixes [this issue](https://github.com/atlas-engineer/nyxt/commit/f6307328626601e7358dee0d2c04c116b2e044fa#commitcomment-88844270).

# Discussion

I took a different approach than that suggested by @aartaka in the aforementioned issue. I find it simpler and more effective.

@aartaka can you confirm that the UI on the REPL's side looks as it should? Thanks.

# Checklist:
Everything in this checklist is required for each PR.  Please do not approve a PR that does not have all of these items.

- [x] I have pulled from master before submitting this PR
- [x] There are no merge conflicts.
- [x] I've added the new dependencies as:
  - [x] ASDF dependencies,
  - [x] Git submodules,
    ```sh
	cd /path/to/nyxt/checkout
    git submodule add https://gitlab.common-lisp.net/nyxt/py-configparser _build/py-configparser
    ```
  - [x] and Guix dependencies.
- [x] My code follows the style guidelines for Common Lisp code. See:
  - [Norvig & Pitman's Tutorial on Good Lisp Programming Style (PDF)](https://www.cs.umd.edu/~nau/cmsc421/norvig-lisp-style.pdf)
  - [Google Common Lisp Style Guide](https://google.github.io/styleguide/lispguide.xml)
- [x] I have performed a self-review of my own code.
- [ ] My code has been reviewed by at least one peer.  (The peer review to approve a PR counts.  The reviewer must download and test the code.)
- [x] Documentation:
  - [x] All my code has docstrings and `:documentation`s written in the aforementioned style.  (It's OK to skip the docstring for really trivial parts.)
  - [x] I have updated the existing documentation to match my changes.
  - [x] I have commented my code in hard-to-understand areas.
  - [x] I have updated the `changelog.lisp` with my changes if it's anything user-facing (new features, important bug fix, compatibility breakage).
  - [x] I have added a `migration.lisp` entry for all compatibility-breaking changes.
- [x] Compilation and tests:
  - [x] My changes generate no new warnings.
  - [x] I have added tests that prove my fix is effective or that my feature works.  (If possible.)
  - [x] New and existing unit tests pass locally with my changes.

### CODE REVIEW

**Comment by aartaka** [comment]
I suggest yet a different of an approach: define .accept per tag type! E.g.:
```  lisp
.accent
:background-color theme:accent
:color theme:on-accent
```
``` lisp
"p.accent, h1.accent, h2.accent ..."
:background-color theme:background
:color theme:accent
```

It preserves the conventional/convenient `.accent` class, while fixing accents for text. And, actually, all accented text elements should better be goreground-accented by default. Whic means: no need to add more classes, just adjust the behavior of existing ones!

**Comment by aadcg** [comment]
You have confused me now! How does that differ for your former suggestion?

**Comment by aadcg** [comment]
Also, say you set the accent class to the heading present in our startup page. Then it will match both the accent class and the accent class relative to `h1`. In short, you'll end up with both the foreground and background covered by the `accent-color`.

**Comment by aadcg** [comment]
To make it clear, apply the following onto master and check the startup page.

```diff
diff --git a/source/buffer.lisp b/source/buffer.lisp
index 870abc89c..c88b007f7 100644
--- a/source/buffer.lisp
+++ b/source/buffer.lisp
@@ -85,6 +85,8 @@ various parts, such as the path of all data files.")
             (.accent
              :background-color theme:accent
              :color theme:on-accent)
+            (h1.accent
+             :color theme:accent)
             (|.button:hover|
              :opacity 0.8)
             (|.button:visited|
```

**Comment by aartaka** [comment]
> You have confused me now! How does that differ from your former suggestion?

It doesn't :P Because using the selectors mechanism that CSS provides is more effective than simulating it manually, especially for such DWIM-ish universal styles as `.accent`.

**Comment by aartaka** [comment]
> To make it clear, apply the following onto master and check the startup page.
> 
> ```diff
> diff --git a/source/buffer.lisp b/source/buffer.lisp
> index 870abc89c..c88b007f7 100644
> --- a/source/buffer.lisp
> +++ b/source/buffer.lisp
> @@ -85,6 +85,8 @@ various parts, such as the path of all data files.")
>              (.accent
>               :background-color theme:accent
>               :color theme:on-accent)
> +            (h1.accent
> +             :color theme:accent)
>              (|.button:hover|
>               :opacity 0.8)
>              (|.button:visited|
> ```

Yes, the background will be accented, because you only redefined `:color` for `h1`. You didn't specifically redefine `:background-color`, as I _explicitly_ suggested in my commentary ｜−・;）

**Comment by aartaka** [comment]
> I disagree with the per element approach. We're dealing with a CSS class that is literally used in 2 spots (REPL and startup page)! 

This can (and will) change. The fact it's used in just two places doesn't mean it won't be used anywhere else. And the fact it's already incomplete hints that it will inevitably be incomplete once we add it anywhere else.

> Why should we generalize something for which we lack experience? To be honest, I even find the use of the accent CSS class backwards. If you take a closer look, we usually define CSS classes for concrete purposes and then we make use of the `accent` color.

Yes, and this pattern is so frequent that it may deserve a CSS class. Compare:
- You define a `.button-accent :color theme:accent` and then use it as `:button.button-accent`, or
- You re-use the existing class with mere `:button.accent`.

> On the other hand, assuming we'd take your approach, I really don't like it because it hardcodes the background, as you have explicitly suggested.

Yes, it does. And that's for good: text elements, even accented ones, should only appear on a text background.

Alternative to the form I suggested it in, you can make `.accent` to be foreground-accented by default, and define a background accented version for e.g. `.button.accent`. Use your judgment of which of the two will make you write less code.

**Comment by aadcg** [comment]
>  Yes, it does. And that's for good: text elements, even accented ones, should only appear on a text background.

Why making such an opinionated decision up front? It sounds reasonable to imagine you'd have a background colored in either `primary` or `secondary` where text elements would be in `accent`.

**Comment by aartaka** [comment]
> > Yes, it does. And that's for good: text elements, even accented ones, should only appear on a text background.
> 
> Why making such an opinionated decision up front? It sounds reasonable to imagine you'd have a background colored in either `primary` or `secondary` where text elements would be in `accent`.

What about the alternative then?

**Comment by aadcg** [comment]
Your suggestion, if I understood correctly, only differs from the one I started this PR with in the sense that I called the classes `accent-foreground` and `accent-background` and you advocate for `accent` and `accent-background`. See the latest commits.

**Comment by aartaka** [comment]
> Your suggestion, if I understood correctly, only differs from the one I started this PR with in the sense that I called the classes `accent-foreground` and `accent-background` and you advocate for `accent` and `accent-background`. See the latest commits.

I'm not advocating for any additional classes, I advocate for dispatching an intuitive `.accent` CSS class over HTML tag names—an ability that CSS gives us from its first versions (҂⌣̀_⌣́) See latest comments in #2626 for an example.

**Comment by aadcg** [comment]
I thought we have agreed that that suggestion is short-sighted since:

> Why making such an opinionated decision up front? It sounds reasonable to imagine you'd have a background colored in either primary or secondary where text elements would be in accent.

**Comment by aadcg** [comment]
I was referring to this when I mentioned i was unsure if I have followed you correctly:

> Alternative to the form I suggested it in, you can make .accent to be foreground-accented by default, and define a background accented version for e.g. .button.accent. Use your judgment of which of the two will make you write less code.

**Comment by aartaka** [comment]
OK, so I was referring to having
``` css
.accent { ... }
p.accent, h1.accent, h2.accent, ... { ... }
```
not 
``` css
.accent { ... }
.accent-background /* or any other name for this second distinct class*/ { ... } 
```

**Comment by aadcg** [comment]
That is clear to me. 

What I don't follow is how you can adopt that approach and avoid hardcoding the background color to `theme:background` since:

> It sounds reasonable to imagine you'd have a background colored in either primary or secondary where text elements would be in accent.

**Comment by aadcg** [comment]
Actually, I may have understood after all.

Check the overall diff of this PR. I agree with this. If you agree too, I'll squash and merge.

**Comment by aartaka** [comment]
> > Alternative to the form I suggested it in, you can make .accent to be foreground-accented by default, and define a background accented version for e.g. .button.accent. Use your judgment of which of the two will make you write less code.
> 
> I think I didn't understand what you meant here.
> 
> It seems quite different from:
> 
> ```css
> .accent { ... }
> p.accent, h1.accent, h2.accent, ... { ... }
> ```

Yes, because it's an alternative to it. To sum up, I suggest TWO options that would work perfectly without introducing new classes and breaking our CSS API:
1. Define per-text-tag `.accent` color the foreground in accent, with non-accent background:
``` css
.accent { ... }
p.accent, h1.accent, h2.accent, ... { ... }
```
2. Or redefine the default `.accent` to be foreground-only (which has the benefit of preserving the background color of the element `.accent` is applied to), while defining the background-accented version for classes and tags that need the background version:
```
.accent {color: accent;}
.button.accent, div.accent ... {background-color: accent; color: on-accent}
```

I'm leaning to the first option, because I feel that accented background is more frequent of a use-case than the foreground one is.

File: source/buffer.lisp [hunk 1]
```diff
@@ -75,6 +75,9 @@ various parts, such as the path of all data files.")
              :border-width "0px"
              :padding "6px"
              :margin "2px")
+            (button.accent
```
Comment (aartaka) [inline source/buffer.lisp:78]: I'd rather do `.button.accent`, because it's not only `<button>`s that we style—it's also `<select>`s and `<details>`.

Comment (aadcg) [reply source/buffer.lisp:78]: Ah, I see!

**Comment by aadcg** [comment]
Woohoo.

Keep in mind, as I am keeping it as well, your suggestion for the future. Nothing is set in stone.

**Pull request merged**